pkg/polaris/patcher.go

```diff
@@ -0,0 +1,192 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package polaris
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	appsv1 "k8s.io/api/apps/v1"
+	appsv1beta1 "k8s.io/api/apps/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func fromYaml(content string, polaris Polaris) (map[string]runtime.Object, error) {
+	// Basic
+	content = strings.ReplaceAll(content, "${NAMESPACE}", polaris.Namespace)
+	content = strings.ReplaceAll(content, "$(JAEGER_NAMESPACE)", polaris.Namespace)
+	content = strings.ReplaceAll(content, "$(EVENT_STORE_NAMESPACE)", polaris.Namespace)
+	content = strings.ReplaceAll(content, "$(LEADER_ELECTOR_NAMESPACE)", polaris.Namespace)
+	content = strings.ReplaceAll(content, "${ENVIRONMENT_NAME}", polaris.Namespace)
+	content = strings.ReplaceAll(content, "${POLARIS_ROOT_DOMAIN}", polaris.EnvironmentDNS)
+	content = strings.ReplaceAll(content, "${IMAGE_PULL_SECRETS}", polaris.ImagePullSecrets)
+	content = strings.ReplaceAll(content, "${INGRESS_CLASS}", polaris.IngressClass)
+
+	// PVC
+	content = strings.ReplaceAll(content, "${DOWNLOAD_SERVER_PV_SIZE}", polaris.PolarisSpec.DownloadServerDetails.Storage.StorageSize)
+	content = strings.ReplaceAll(content, "${EVENTSTORE_PV_SIZE}", polaris.PolarisDBSpec.EventstoreDetails.Storage.StorageSize)
+	content = strings.ReplaceAll(content, "${MONGODB_PV_SIZE}", polaris.PolarisDBSpec.MongoDBDetails.Storage.StorageSize)
+	content = strings.ReplaceAll(content, "${UPLOAD_SERVER_PV_SIZE}", polaris.PolarisDBSpec.UploadServerDetails.Storage.StorageSize)
+	content = strings.ReplaceAll(content, "${POSTGRES_PV_SIZE}", polaris.PolarisDBSpec.PostgresDetails.Storage.StorageSize)
+	content = strings.ReplaceAll(content, "${REPORT_STORAGE_PV_SIZE}", polaris.ReportingSpec.ReportStorageDetails.Storage.StorageSize)
+
+	// SMTP
+	content = strings.ReplaceAll(content, "${SMTP_SENDER_EMAIL}", polaris.PolarisDBSpec.SMTPDetails.SenderEmail)
+	content = strings.ReplaceAll(content, "${SMTP_HOST}", polaris.PolarisDBSpec.SMTPDetails.Host)
+	if polaris.PolarisDBSpec.SMTPDetails.Port != 2525 {
+		// TODO this needs to be a placeholder
+		content = strings.ReplaceAll(content, "2525", strconv.Itoa(polaris.PolarisDBSpec.SMTPDetails.Port))
+	}
+	if len(polaris.PolarisDBSpec.SMTPDetails.Username) != 0 {
+		content = strings.ReplaceAll(content, "${SMTP_USERNAME}", EncodeStringToBase64(polaris.PolarisDBSpec.SMTPDetails.Username))
+	} else {
+		content = strings.ReplaceAll(content, "${SMTP_USERNAME}", "Cg==")
+	}
+	if len(polaris.PolarisDBSpec.SMTPDetails.Password) != 0 {
+		content = strings.ReplaceAll(content, "${SMTP_PASSWORD}", fmt.Sprintf("\"%s\"", EncodeStringToBase64(polaris.PolarisDBSpec.SMTPDetails.Password)))
+	} else {
+		content = strings.ReplaceAll(content, "${SMTP_PASSWORD}", "Cg==")
```

Comment:
confused about encoding of this now, is it needed?...since we're now storing the entire config as a secret anyway

Reply:
`polaris.PolarisDBSpec.SMTPDetails.Password` is not base64 encoded and we are replacing the value inside the yaml file and not a v1.Secret struct so it needs to be base64 encoded.